pkg/thing/interactors/update_data.go

```diff
@@ -0,0 +1,91 @@
+package interactors
+
+import (
+	"fmt"
+
+	"github.com/CESARBR/knot-babeltower/pkg/network"
+	"github.com/CESARBR/knot-babeltower/pkg/thing/entities"
+)
+
+var (
+	// ErrNoAuthToken error occur when no authorization token is provided
+	ErrNoAuthToken = fmt.Errorf("missing authorization token")
+
+	// ErrNoIDParam error occur when no thing's id is provided
+	ErrNoIDParam = fmt.Errorf("missing thing's id")
+
+	// ErrNoDataParam error occur when no thing's data is provided
+	ErrNoDataParam = fmt.Errorf("missing thing's data")
+
+	// ErrNoSchema error occur when the thing has no schema yet
+	ErrNoSchema = fmt.Errorf("thing has no schema")
+
+	// ErrDataInvalid error occur when the provided data mismatch the thing's schema
+	ErrDataInvalid = fmt.Errorf("data is incompatible with thing's schema")
+)
+
+// UpdateData executes the use case operations to update data in thing
+func (i *ThingInteractor) UpdateData(authorization, thingID string, data []network.Data) error {
+	err := i.verifyThingData(authorization, thingID, data)
+	if err != nil {
+		return fmt.Errorf("unable to update data: %w", err)
+	}
+
+	err = i.clientPublisher.SendUpdateData(thingID, data)
+	if err != nil {
+		return fmt.Errorf("error sending message to client: %w", err)
+	}
+
+	i.logger.Info("data update command successfully sent")
+	return nil
+}
+
+func (i *ThingInteractor) verifyThingData(authorization, thingID string, data []network.Data) error {
+	if authorization == "" {
```

Comment:
You're doing more than verifying thing's data in this function, right?

Reply:
@joaoaneto It's true. So, I replace the parameters check to main `UpdateData` function. Done!